llvm/lib/Target/AMDGPU/AMDGPUUniformIntrinsicCombine.cpp

```diff
@@ -97,14 +91,12 @@ static bool optimizeUniformIntrinsic(IntrinsicInst &II,
           Tracker[NotOp] = true; // NOT preserves uniformity
           LLVM_DEBUG(dbgs() << "Replacing ICMP_EQ: " << *NotOp << '\n');
           ICmp->replaceAllUsesWith(NotOp);
-          ICmp->eraseFromParent();
```

Comment:
I vaguely remember that since you use early_inc_range in the outer for-loop, it should be safe to call this erase, right?

Reply:
SGTM, but I don't know if that is the usual practice for LLVM in general.